packages/components/src/Grid.astro

```diff
@@ -0,0 +1,219 @@
+---
+const { container, fluid, xs, sm, md, lg, auto, gap } = Astro.props;
+const containerClass = container ? "grid-container" : "";
+const autoClass = auto ? "auto-columns" : "";
+const gapClass = gap ? `gap-${gap}` : "";
+const xsClass = xs ? `col-${xs}` : "";
+const smClass = sm ? `col-sm-${sm}` : "";
+const mdClass = md ? `col-md-${md}` : "";
+const lgClass = lg ? `col-lg-${lg}` : "";
+const fluidClass = fluid ? "fluid" : "";
+const classList = [
+  containerClass,
+  fluidClass,
+  autoClass,
+  gapClass,
+  xsClass,
+  smClass,
+  mdClass,
+  lgClass,
+]
+  .join(" ")
+  .trim();
+---
+
+<style lang="scss">
+  .grid-container {
+    display: grid;
+    grid-template-columns: repeat(12, 1fr);
+    gap: 16px;
+    max-width: 1440px;
+    margin: 0 auto;
+    padding: 0 16px;
+
+    &.fluid {
+      max-width: 100%;
+    }
+
+    @media (min-width: 768px) {
+      padding: 0 32px;
+    }
+  }
+
+  .auto-columns {
+    grid-template-columns: repeat(auto-fill, minmax(min-content, 1fr));
+  }
+
+  .gap-1 {
+    gap: 4px;
```

Comment:
Not one for this PR, but at some point we should consider moving these to variables which are re-used for all our components. This helps ensure consistency but also we can do this in such a way that consumers can override them (e.g. say they want to go with 5, 10, 15, 20 instead of 4, 8, 16, 24).